src/main/java/org/elasticsearch/search/aggregations/reducers/derivative/DerivativeBuilder.java

```diff
@@ -45,6 +46,16 @@ public DerivativeBuilder gapPolicy(GapPolicy gapPolicy) {
         return this;
     }
 
+    public DerivativeBuilder units(String units) {
+        this.units = units;
+        return this;
+    }
+
+    public DerivativeBuilder units(DateHistogramInterval units) {
+        this.units = units.toString();
+        return this;
+    }
```

Comment:
So String method is also required for units like "5d" or "1.5w" so we can parse them into a time period. Sorry I hadn't remembered this before.